pkg/generators/markers_test.go

```diff
@@ -131,20 +131,102 @@ func TestParseCommentTags(t *testing.T) {
 		},
 		{
 
-			t:    &structKind,
+			t:    structKind,
 			name: "invalid: invalid value",
 			comments: []string{
 				"+k8s:validation:",
 			},
 			expectedError: `failed to parse marker comments: cannot have empty key for marker comment`,
 		},
 		{
-			t:    &numKind,
+			t: types.Float64,
+			// temporary test. ref support may be added in the future
 			name: "ignore refs",
 			comments: []string{
 				"+k8s:validation:pattern=ref(asdf)",
 			},
 		},
+		{
+			t:    types.Float64,
+			name: "cel rule",
+			comments: []string{
+				`+k8s:validation:cel[0]:rule="oldSelf == self"`,
+				`+k8s:validation:cel[0]:message="immutable field"`,
+			},
+			expected: generators.CommentTags{
+				CEL: []generators.CELTag{
+					{
+						Rule:    "oldSelf == self",
+						Message: "immutable field",
+					},
+				},
+			},
+		},
+		{
+			t:    types.Float64,
+			name: "skipped CEL rule",
+			comments: []string{
+				`+k8s:validation:cel[0]:rule="oldSelf == self"`,
+				`+k8s:validation:cel[0]:message="immutable field"`,
+				`+k8s:validation:cel[2]:rule="self > 5"`,
+				`+k8s:validation:cel[2]:message="must be greater than 5"`,
+			},
+			expectedError: `invalid marker comments: empty CEL tag is not allowed`,
+		},
+		{
+			t: types.Float64,
+			// This should parse, but return an error in validation since
```

Comment:
Is this comment referring to the previous `skipped CEL rule`?

Reply:
Yeah, moved it